src/main/php/org/bovigo/vfs/visitor/vfsStreamAbstractVisitor.php

```diff
@@ -29,24 +31,16 @@ abstract class vfsStreamAbstractVisitor implements vfsStreamVisitor
      */
     public function visit(vfsStreamContent $content): vfsStreamVisitor
     {
-        switch ($content->getType()) {
-            case vfsStreamContent::TYPE_BLOCK:
-                $this->visitBlockDevice($content);
-                break;
-
-            case vfsStreamContent::TYPE_FILE:
-                $this->visitFile($content);
-                break;
-
-            case vfsStreamContent::TYPE_DIR:
-                if (!$content->isDot()) {
-                    $this->visitDirectory($content);
-                }
-
-                break;
-
-            default:
-                throw new \InvalidArgumentException('Unknown content type ' . $content->getType() . ' for ' . $content->getName());
+        if ($content instanceof vfsStreamBlock) {
+            $this->visitBlockDevice($content);
+        } elseif ($content instanceof vfsStreamFile) {
+            $this->visitFile($content);
+        } elseif ($content instanceof vfsStreamDirectory) {
+            if (!$content->isDot()) {
+                $this->visitDirectory($content);
+            }
+        } else {
+            throw new \InvalidArgumentException('Unknown content type ' . $content->getType() . ' for ' . $content->getName());
         }
```

Comment:
What‘s the reasoning for replacing the switch/case block with an if/elseif block?

Reply:
It's mainly so phpstan can verify what each `$content` actually is, as this was something it identified as having errors. The `visit()` accepts any `vfsStreamContent`, but all the methods it calls internally require a specific implementation of `vfsStreamContent` and will blow up if anything else is passed to them.

So if a PR like https://github.com/bovigo/vfsStream/pull/156 introduced a `vfsErroneousFile` and identified it as `TYPE_FILE`, the switch statement would have handed it off to `visitFile()` and then an error would have been throw. That is, unless `vfsErroneousFile` extended `vfsStreamFile`.

The `instanceof` check should verify that each method being called for sure accepts the `$content` we're passing to it.

This is also why I love phpstan because it can catch potential bugs before they're ever an issue.

Reply:
I still don't like it, but can't really explain why. So no further objection on this.